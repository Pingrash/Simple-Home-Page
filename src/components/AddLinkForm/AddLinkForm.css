.close-button {
  align-self: flex-end;
  margin: 1.5rem 0 0 0;
  flex: 1;
  font-size: 1rem;
  border-radius: 20%;
  cursor: pointer;
}

.add-form {
  display: flex;
  flex-direction: column;
  flex: 3;
  padding: 0 1rem 2rem;
}

.submit-button {
  margin-top: 1rem;
  background-color: #0b3c97;
  color: #f1f1f1;
  font-size: 1rem;
  padding: 0.5rem 0;
}

.submit-button:hover {
  background-color: #18a2cf;
}

.message {
  flex: 1;
  align-self: center;
  padding-bottom: 1rem;
  font-size: 1.5rem;
}
