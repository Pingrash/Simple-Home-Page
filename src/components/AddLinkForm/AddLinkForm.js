import React, { useState } from 'react';
import './AddLinkForm.css';
import PropTypes from 'prop-types';

const AddLinkForm = ({ addLink }) => {
  const [url, updateUrl] = useState('');
  const [linkName, updateLinkName] = useState('');
  const [linkAdded, updateLinkAdded] = useState(false);

  return (
    <>
      <form
        onSubmit={e => {
          e.preventDefault();
          const newLink = {
            link: url,
            favicon: `http://www.google.com/s2/favicons?domain=${url}`,
            name: linkName
          };
          addLink(newLink);
          updateLinkAdded(true);
        }}
        className='add-form'
        data-test='entry-form'
      >
        <h3>Add Link</h3>
        <label for='url'>URL: </label>
        <input
          id='url'
          type='url'
          onChange={e => updateUrl(e.target.value)}
          title='Website URL eg. https://www.google.com'
          required
          placeholder='https://'
          defaultValue='https://'
          data-test='form-field'
        />

        <label for='name'>Link Name: </label>
        <input
          id='name'
          onChange={e => updateLinkName(e.target.value)}
          required
          data-test='form-field'
        />
        <button
          type='submit'
          className='submit-button'
          data-test='submit-button'
        >
          Submit
        </button>
      </form>
      {linkAdded ? (
        <span className='success message'>Link Added</span>
      ) : null}
    </>
  );
};

AddLinkForm.propTypes = {
  addLink: PropTypes.func.isRequired
};

export default AddLinkForm;
