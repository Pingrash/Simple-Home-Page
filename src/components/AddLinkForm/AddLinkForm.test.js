import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { findByDataTest, checkProps } from '../../../Utilities';
import AddLinkForm from './AddLinkForm';

const testFunc = () => null;

const props = {
  addLink: testFunc
};

const setup = () => {
  const component = shallow(<AddLinkForm {...props} />);
  return component;
};

describe('Snapshot', () => {
  const tree = renderer.create(<AddLinkForm {...props} />).toJSON();

  expect(tree).toMatchSnapshot();
});

describe('AddLinkForm renders correctly', () => {
  let wrapper;
  beforeEach(() => (wrapper = setup()));

  it('Should render without errors', () => {
    expect(wrapper.length).toBe(1);
  });

  it('Should render two input fields', () => {
    const fields = findByDataTest(wrapper, 'form-field');
    expect(fields.length).toBe(2);
  });

  it('Should render a submit button', () => {
    const button = findByDataTest(wrapper, 'submit-button');
    expect(button.length).toBe(1);
  });
});

describe('AddLinkForm has correct props', () => {
  const propsError = checkProps(AddLinkForm, props);
  expect(propsError).toBeUndefined();
});
