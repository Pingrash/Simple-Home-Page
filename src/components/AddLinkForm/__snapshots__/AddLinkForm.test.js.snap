// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<form
  className="add-form"
  data-test="entry-form"
  onSubmit={[Function]}
>
  <h3>
    Add Link
  </h3>
  <label
    for="url"
  >
    URL: 
  </label>
  <input
    data-test="form-field"
    defaultValue="https://"
    id="url"
    onChange={[Function]}
    placeholder="https://"
    required={true}
    title="Website URL eg. https://www.google.com"
    type="url"
  />
  <label
    for="name"
  >
    Link Name: 
  </label>
  <input
    data-test="form-field"
    id="name"
    onChange={[Function]}
    required={true}
  />
  <button
    className="submit-button"
    data-test="submit-button"
    type="submit"
  >
    Submit
  </button>
</form>
`;
